crates/bdk/src/wallet/utils.rs

```diff
@@ -110,6 +111,19 @@ impl<Pk: MiniscriptKey + ToPublicKey> Satisfier<Pk> for Older {
     }
 }
 
+// The Fisher-Yates shuffling algorithm to randomly order a list of N items: [https://en.wikipedia.org/wiki/Fisher%E2%80%93Yates_shuffle]
```

Comment:
totally unimportant suggestion:

```suggestion
// The Knuth shuffling algorithm based on the original Fisher-Yates method: [https://en.wikipedia.org/wiki/Fisher%E2%80%93Yates_shuffle]
```

Also, wouldn't hurt to have a unit test for `shuffle_slice`.